packages/babel-preset-gatsby/src/optimize-hook-destructuring.ts

```diff
@@ -52,8 +52,14 @@ export default function ({
       if (!(onlyBuiltIns ? isBuiltInHook : isHook).test(hookName)) return
 
       path.parent.id = t.objectPattern(
-        path.parent.id.elements.map((element, i) =>
-          t.objectProperty(t.numericLiteral(i), element!)
+        path.parent.id.elements.reduce(
```

Comment:
not very happy with using `.reduce` here ... but TS was not very happy with 
`.map().filter(Boolean)` trick or other similar ones (it still would still complain that I try to use array of `BabelTypes.ObjectProperty | null` as param of `t.objectPattern`

Reply:
I think something like this could work:

```
.filter(<T>(n: T | null | undefined): n is T => Boolean(n))
```

But not sure if it is easier to read :)

Reply:
Happy to try it, but honestly I'm not sure if it's actually more readable - especially for someone that isn't very "fluent" in typescript (so for example me :P)

Reply:
I'll skip this doing any changes - it doesn't seem that impactful one way or the other, and we can better spend our time :)